daemon/api_aspects.go

```diff
@@ -53,9 +55,22 @@ func getAspect(c *Command, r *http.Request, _ *auth.UserState) Response {
 	st.Lock()
 	defer st.Unlock()
 
+	_, err := aspectstate.GetDatabag(st, account, bundleName)
+	if err != nil {
+		if errors.Is(err, state.ErrNoState) {
+			return toAPIError(&aspects.AspectNotFoundError{Account: account, BundleName: bundleName, Aspect: aspect})
+		}
+		return toAPIError(err)
+	}
```

Comment:
do we need GetDatabag and this code, won't NewTransaction below end up reading the databag as well and error? could we move the check to just there?

Reply:
The transaction's databag reader creates the databag if it can't find any (since it's also used on writes) so it won't return an AspectNotFound at transaction creation. But yes, the created databag will be empty so further down the line it returns a FieldNotFound error. 

That reminds me, I think there's isn't much functional difference between the not found errors anymore so, in a follow-up PR, I can see about merging them so the error messages are consistent (regardless of what level they occurred)